### ISSUE TITLE
Building sphinx-docs from source gives warning, "Inline emphasis start-string without end-string."

### ISSUE THREAD

**Issue by mojotx** [fish-shell/fish-shell#7519]
When building from source (3.1.2-1706-geb592092a) on macOS Big Sur, I see the following warnings with the `sphinx-docs` target:

```
FISH_BUILD_VERSION=3.1.2-1706-geb592092a
[  0%] Built target CHECK-FISH-BUILD-VERSION-FILE
Scanning dependencies of target fishlib
[  1%] Building CXX object CMakeFiles/fishlib.dir/src/fish_version.cpp.o
[  1%] Linking CXX static library libfishlib.a
[ 96%] Built target fishlib
[ 97%] Linking CXX executable fish_indent
[ 98%] Built target fish_indent
[100%] Building HTML documentation with Sphinx
../doc_src/cmds/string-match.rst:13: WARNING: Inline emphasis start-string without end-string.
../doc_src/cmds/string-match.rst:13: WARNING: Inline emphasis start-string without end-string.
/Users/mjarvis/src/github.com/fish-shell-mojotx/doc_src/cmds/string-match.rst:30: WARNING: Inline emphasis start-string without end-string.
/Users/mjarvis/src/github.com/fish-shell-mojotx/doc_src/cmds/string-match.rst:30: WARNING: Inline emphasis start-string without end-string.
[100%] Built target sphinx-docs
```

**Comment by mqudsi** [comment]
That's on me; I tested with `pandoc` to verify that I could have italics ending within the middle of a word and was satisfied with the results.

If anyone can shed light how to write the following:

> some words then *n*th object

with only the first letter of "nth" being italicized.

**Comment by charego** [comment]
@mqudsi some suggestions https://github.com/fish-shell/fish-shell/pull/7520#discussion_r533095541

The superscript one could be tidied up with some css I assume.

**Comment by mojotx** [comment]
I like @zanchey's [idea](https://github.com/fish-shell/fish-shell/pull/7520#issuecomment-736252564) of simply rewording the text to eliminate the need of having a word partially italicized.

**Comment by ridiculousfish** [comment]
This got fixed in eb22a9c4db1a2b7ffda030650ca00a393829b03b
